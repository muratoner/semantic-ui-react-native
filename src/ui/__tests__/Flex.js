import {shallow} from 'enzyme'
import toJson from 'enzyme-to-json'
import React from 'react'
import {StyleSheet} from 'react-native'
import {Avatar, Flex} from '../../'

describe('Flex Component', () => {
	it('should render without issues', () => {
		const component = shallow(<Flex style={{backgroundColor: 'red'}} />)

		expect(component.length).toBe(1)
		const style = StyleSheet.flatten(component.get(0).props.style)
		expect(style).toHaveProperty('flex', 1)
		expect(style).toHaveProperty('backgroundColor', 'red')
		expect(toJson(component)).toMatchSnapshot()
	})

	it('should render with 2 flex size', () => {
		const component = shallow(<Flex size={2} />)

		expect(component.length).toBe(1)
		const style = StyleSheet.flatten(component.get(0).props.style)
		expect(style).toHaveProperty('flex', 2)
		expect(toJson(component)).toMatchSnapshot()
	})

	it('should render with two children', () => {
		const component = shallow(
			<Flex>
				<Avatar title='Murat Ã–NER' />
				<Avatar title='Murat' />
			</Flex>
		)

		expect(component.length).toBe(1)
		expect(component.get(0).props.children.length).toBe(2)
		expect(toJson(component)).toMatchSnapshot()
	})
})
