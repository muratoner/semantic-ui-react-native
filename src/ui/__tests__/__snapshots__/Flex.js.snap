// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Flex Component should render with 2 flex size 1`] = `
<View
  style={
    Array [
      Object {
        "flex": 2,
      },
      undefined,
    ]
  }
/>
`;

exports[`Flex Component should render with two children 1`] = `
<View
  style={
    Array [
      Object {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <Component
    title="Murat Ã–NER"
  />
  <Component
    title="Murat"
  />
</View>
`;

exports[`Flex Component should render without issues 1`] = `
<View
  style={
    Array [
      Object {
        "flex": 1,
      },
      Object {
        "backgroundColor": "red",
      },
    ]
  }
/>
`;
